### PR TITLE
change the way to get right FP gacha subId

### PR DESCRIPTION
**Pull request by Zhen-Bo**
# how it works
Objects with commonReleaseId within the same level of high priority are checked first.
If all conditions of the high priority are present but not met, then the next priority level is checked.
The only condition that is currently being checked is 'clearQuest' and 'clearNum'

# example
Right now, we've got three subIds activated: 402, 403, and 1000. 
The function first checks the highest priority, which is 403. But there's a commonReleaseId and it's condition is 'Clear LB6'.

If your account doesn't clear LB6, it'll move on to check the lower priorities, which are 402 and 1000.

Now, if there are multiple subIds with the same priority, it'll prioritize the one with the commonReleaseId first, like 1000 in this case. But since its condition is 'Clear LB6', it'll then check the one without a commonReleaseId , which is 402.

notes
---
剛好最近重構到友抽的部分, 看到這邊你抽了 funtion 感覺可以發個 RP

結果抽 function 後改起來的體驗也沒好到哪裡去
看源碼時如果感到窒息就可以吸一口氧的話
我估計能把方圓 20 里內醫院裡的氧氣鋼瓶全吸光

已經盡量保證正確性的情況下對源碼做最小改動, 畢竟在我的 IDE 開啟專案後
只要一按保存就會瞬間對一堆東西做自動修正...

只開三個檔案就出了 118 個 error+warning....
image

### CODE REVIEW

**Comment by DNNDHH** [comment]
池子是活动池优先，按现在的池子…没有过章新号在没有活动时候是0或401…有活动卡池时候是402…而正常已经过章完主线的账号是403

File: libs/GetSubGachaId.py [hunk 1]
```diff
@@ -1,22 +1,53 @@
...
+        # please implement webhook sned error or etc here
+        pass
+
+    # Start by sorting from highest to lowest priority, then within the same priority, sort by commonReleaseId
+    sortedGachaList = sorted(actiaveGacha, key=lambda x: (x["priority"], x["commonReleaseId"]), reverse=True)
+    for gacha in sortedGachaList:
+        if gacha["commonReleaseId"] != 0 and isMatchedQuestCondition(region, userQuest, gacha["commonReleaseId"]):
+            return gacha["id"]
+        elif gacha["commonReleaseId"] == 0:
+            return gacha["id"]
```
Comment (Zhen-Bo) [inline libs/GetSubGachaId.py:53]: 我看 401 的 close at 已經超過現在的時間點了
所以如果以後沒有活動的話 EX: 402, 403 都關閉了, 只剩下 1000 還開啟
但是又有帳號還沒過 LB6 的話, 他就只能抽 subId 為 0 的池嗎?
如果是的話可能 那這邊的 for 迴圈底下就是要在加個 return 0

Comment (Zhen-Bo) [inline libs/GetSubGachaId.py:53]: 我看 401 的 close at 已經超過現在的時間點了
所以如果以後沒有活動的話 EX: 402, 403 都關閉了, 只剩下 1000 還開啟
但是又有帳號還沒過 LB6 的話, 他就只能抽 subId 為 0 的池嗎?
如果是的話可能 那這邊的 for 迴圈底下就是要在加個 return 0

Comment (Zhen-Bo) [reply libs/GetSubGachaId.py:53]: 這邊應該是說如果 49 行這層 for 迴圈跑完後如果都沒有符合的條件就要返回 0
抱歉平時都是用 gitlab 不太熟悉 github

**Pull request closed without merging**